File: 1442.cpp
using namespace std;
class Solution {
public:
    int pain(int p,vector<int> s){
        int tar=0;
        for(int i=0;i<p;i++){
            tar^=s[i];
        }
            return tar;
        };
    int countTriplets(vector<int>& arr) {
        int count=0;
        int n=size(arr);
        for(int i=0;i<n;i++){
            for(int k=i;k<n;k++){
                if (pain(i,arr)==pain(k+1,arr)){
                    count+=k-i;
                }
            }
        }
        return count;
    }
};

File: 1442.py
class Solution(object):
    def countTriplets(self, arr):
        n=len(arr)
        def bior(string1):
            x=0
            for i in string1:
                x^=i
            return x
        count=0
        for i in range(0,n):
                for k in range(i+1,n):
                    if bior(arr[:i])^bior(arr[:k+1])==0:
                        count+=k-i
        return count

